client/src/components/Navigation/Navigation.tsx

```diff
@@ -32,6 +32,9 @@ interface Props {
 const Navigation: React.FC<Props> = ({ user, setUser }) => {
   // Sidebar state is being used to see whether sidebar is open or not
   const [sidebar, setSidebar] = useState<boolean>(false);
+  const [path, setPath] = useState<string>(window.location.pathname);
```

Comment:
I trust your judgment, but is there a way to do this without state and a useEffect?

Just wondering.

Reply:
Hmm I tried to do it without, but it won't re-render the navigation bar to apply those changes
If you have any suggestions, let me know and I can always implement it